api/src/main/java/org/openmrs/module/commonreports/reports/AntenatalReportManager.java

```diff
@@ -0,0 +1,378 @@
+				gestationDuration.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+				gestationDuration.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+				gestationDuration.setValue1(Double.parseDouble(firstNumber));
+				gestationDuration.setValue2(Double.parseDouble(lastNumber));
+				gestationDuration.setOperator1(RangeComparator.GREATER_EQUAL);
+				gestationDuration.setOperator2(RangeComparator.LESS_EQUAL);
+				
+				antenatalGestation.addRow(
+				    member + " " + MessageUtil.translate("commonreports.report.antenatalGestation.gestationWeeks"),
+				    gestationDuration, parameterMappings);
+				
+			}
+			
+		}
+		
+		//Risks related
+		GenderCohortDefinition female = new GenderCohortDefinition();
+		female.setFemaleIncluded(true);
+		
+		// Risky Pregnancies
+		CodedObsCohortDefinition riskyPregnancy = new CodedObsCohortDefinition();
+		riskyPregnancy.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		riskyPregnancy.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		riskyPregnancy.setOperator(SetComparator.IN);
+		
+		riskyPregnancy.setQuestion(inizService.getConceptFromKey("report.antenatal.riskyPregnancy"));
+		riskyPregnancy.setValueList(Arrays.asList(inizService.getConceptFromKey("report.antenatal.yes")));
+		antenatalRisks.addRow(MessageUtil.translate("commonreports.report.antenatalRisks.riskyPregnancy"), riskyPregnancy,
+		    parameterMappings);
+		
+		// Iron def ANC visit Pregnancies
+		CodedObsCohortDefinition anemia = new CodedObsCohortDefinition();
+		anemia.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		anemia.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		anemia.setOperator(SetComparator.IN);
+		anemia.setQuestion(inizService.getConceptFromKey("report.antenatal.codedDiagnosis"));
+		anemia.setValueList(Arrays.asList(inizService.getConceptFromKey("report.antenatal.anemiaIronDeficiency")));
+		
+		VisitCohortDefinition _prenatal = new VisitCohortDefinition();
+		_prenatal.setVisitTypeList(Arrays.asList(Context.getVisitService()
+		        .getVisitTypeByUuid(inizService.getValueFromKey("report.antenatal.prenatalVisitType"))));
+		CompositionCohortDefinition ccd = new CompositionCohortDefinition();
+		ccd.initializeFromElements(_prenatal, anemia, female);
+		antenatalRisks.addRow(MessageUtil.translate("commonreports.report.antenatalRisks.ironDefANC"), ccd,
+		    parameterMappings);
+		
+		// Prenatal visit + Fer Folate Co prescribed 
+		SqlCohortDefinition sqd = new SqlCohortDefinition(
+		        "select patient_id from orders where concept_id=(select DISTINCT concept_id from concept where uuid = '"
+		                + inizService.getValueFromKey("report.antenatal.ferrousFolate")
+		                + "' and retired = 0) and order_type_id =(select order_type_id from order_type where uuid = '"
+		                + inizService.getValueFromKey("report.antenatal.drugOrder") + "')");
```

Comment:
Let's therefore add something like `... AND date_created BETWEEN :onOrAfter AND : onOrBefore` to the query

Reply:
Isn't this handled in the following line 204

Reply:
The lines you are referring to simply define the parameters to substitute and until those parameters are present in the sql query, they are not replaced. See here for example.

Reply:
Done